metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizer.java

```diff
@@ -129,23 +151,24 @@ public synchronized void loadSnapshot(Map<Uuid, StandardAcl> acls) {
     public List<AuthorizationResult> authorize(
             AuthorizableRequestContext requestContext,
             List<Action> actions) {
-        StandardAuthorizerData curData = data;
-        List<AuthorizationResult> results = new ArrayList<>(actions.size());
-        for (Action action: actions) {
-            AuthorizationResult result = curData.authorize(requestContext, action);
-            results.add(result);
-        }
-        return results;
+        return inReadLock(() -> {
```

Comment:
I know it does not look as pretty, but perhaps we should just do the `try/finally` blocks. Especially for the case of `authorize`, it is annoying to have the additional allocations just to pass the calback.

Reply:
Okay. I'll remove the functions.